### PR TITLE
publish metadata for all imports

### PR DESCRIPTION
**Pull request by mcovarr**
Still looking at a Centaur test for this but I thought I'd let the seafowl have at it. Metadata looks like:

```
{
  "workflowName": "three_step",
  "submittedFiles": {
    "workflow": "import \"3step-tasks.wdl\" as tasks\nimport \"https://raw.githubusercontent.com/broadinstitute/centaur/591beaf8422af7c3faf51e437a91d94d13b76eba/src/main/resources/standardTestCases/aliased_subworkflows/subworkflow.wdl\" as subworkflow\n\n\nworkflow three_step {\n  call tasks.ps as ps\n  call tasks.cgrep as cgrep {\n    input: in_file = ps.procs\n  }\n  call tasks.wc as wc {\n    input: in_file = ps.procs\n  }\n  output {\n    cgrep.*\n    wc.*\n  }\n}\n",
    "workflowType": "WDL",
    "options": "{\n\n}",
    "inputs": "{\"three_step.cgrep.pattern\":\"mcovarr\"}",
    "labels": "{}",
    "imports": {
      "https://raw.githubusercontent.com/broadinstitute/centaur/591beaf8422af7c3faf51e437a91d94d13b76eba/src/main/resources/standardTestCases/aliased_subworkflows/subworkflow.wdl": "task increment {\n  Int i\n  command {\n    echo $(( ${i} + 1 ))\n  }\n  output {\n    Int j = read_int(stdout())\n  }\n  runtime {\n    docker: \"ubuntu:latest\"\n  }\n}\n\nworkflow subwf {\n  Array[Int] is\n  scatter (i in is) {\n    call increment { input: i = i }\n  }\n  output {\n    Array[Int] js = increment.j\n  }\n}\n",
      "3step-tasks.wdl": "task ps {\n  command {\n    ps\n  }\n  output {\n    File procs = stdout()\n  }\n  runtime {\n    docker: \"ubuntu:latest\"\n  }\n}\n\ntask cgrep {\n  String pattern\n  File in_file\n  command {\n    grep '${pattern}' ${in_file} | wc -l\n  }\n  output {\n    Int count = read_int(stdout())\n  }\n  runtime {\n    docker: \"ubuntu:latest\"\n  }\n}\n\ntask wc {\n  File in_file\n  command {\n    cat ${in_file} | wc -l\n  }\n  output {\n    Int count = read_int(stdout())\n  }\n  runtime {\n    docker: \"ubuntu:latest\"\n  }\n}\n\n"
    }
  },
.
.
.
```

So the http imports come in under "submittedFiles" which is maybe a little weird. But other options would have http imports either not being next to the file imports and/or having to change the metadata schema.

### CODE REVIEW

**Comment by geoffjentry** [comment]
Speaking just for this particular bird of the ocean I like that structure

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 1]
```diff
@@ -475,22 +438,56 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
...
+      }
+
+      // Descend the namespace looking for imports and construct `MetadataEvent`s for them.
+      def collectImportEvents(ns: WdlNamespace): List[MetadataEvent] = {
+        // Do not publish import events for URIs which correspond to literal strings as these are the top-level
+        // submitted workflow.
+        val nonTopLevelEvents = ns.importUri.toList collect {
+          case uri if uri != WdlNamespace.WorkflowResourceString => metadataEventForImportedNamespace(ns)
         }
+        nonTopLevelEvents ++ (ns.namespaces.toList map metadataEventForImportedNamespace)
```
Comment (Horneth) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: Does this cover recursive imports ?

Comment (mcovarr) [reply engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: An earlier version of this did but I don't think this one does. 😦 Good catch!

**Comment by kcibul** [comment]
I like the metadata this way also

**Pull request merged**